publish_lambda.sh

```diff
@@ -0,0 +1,522 @@
+#!/usr/bin/env bash
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+
+set -e
+
+echo "    AWS CLI (https://aws.amazon.com/cli/) SAM (https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/install-sam-cli.html) and Python3.9 with pip3 required"
+echo "    Please, before launching the tool execute \"$ pip3 install ruamel.yaml\""
+
+if [[ $# -ne 5 ]]
+then
+    echo "Usage: $0 config-path lambda-name forwarder-tag bucket-name region"
+    echo "    Arguments:"
+    echo "    config-path: full path to the publish configuration"
+    echo "    lambda-name: name of the lambda to be published in the account"
+    echo "    forwarder-tag: tag of the elastic serverless forwarder to publish"
+    echo "    bucket-name: bucket name where to store the zip artifact for the lambda"
+    echo "                 (it will be created if it doesn't exists, otherwise "
+    echo "                  you need already to have proper access to it)"
+    echo "    region: region where to publish in"
+    exit 1
+fi
+
+PUBLISH_CONFIG="$1"
+LABDA_NAME="$2"
+TAG_NAME="$3"
+BUCKET="$4"
+REGION="$5"
+
+TMPDIR=$(mktemp -d /tmp/publish.XXXXXXXXXX)
+CLONED_FOLDER="${TMPDIR}/sources"
+PACKAGE_FOLDER="${CLONED_FOLDER}/package"
+GIT_REPO="https://github.com/elastic/elastic-serverless-forwarder.git"
+
+trap 'rm -rf ${TMPDIR}' EXIT
+
+aws s3api get-bucket-location --bucket "${BUCKET}" || aws s3api create-bucket --acl private --bucket "${BUCKET}" --region "${REGION}" --create-bucket-configuration LocationConstraint="${REGION}"
+
+mkdir -v -p "${CLONED_FOLDER}"
+git clone --depth 1 --branch "${TAG_NAME}" "${GIT_REPO}" "${CLONED_FOLDER}"
```

Comment:
What's the reasoning behind introducing `TAG_NAME` ? Aren't we always considering `main` as the source of truth?